dbms/src/Functions/FunctionsJson.h

```diff
@@ -1514,4 +1514,400 @@ class FunctionJsonDepth : public IFunction
+        size_t rows = block.rows();
+        if (const auto * col_from = checkAndGetColumn<ColumnString>(column.get()))
+        {
+            const auto & data_from = col_from->getChars();
+            const auto & offsets_from = col_from->getOffsets();
+
+            auto col_to = ColumnVector<UInt8>::create(rows, 0);
+            auto & data_to = col_to->getData();
+
+            size_t current_offset = 0;
+            for (size_t i = 0; i < rows; ++i)
+            {
+                size_t next_offset = offsets_from[i];
+                size_t data_length = next_offset - current_offset - 1;
+                bool is_valid = checkJsonValid(reinterpret_cast<const char *>(&data_from[current_offset]), data_length);
+                data_to[i] = is_valid ? 1 : 0;
+                current_offset = next_offset;
+            }
+
+            block.getByPosition(result).column = std::move(col_to);
+        }
+        else
+            throw Exception(
+                fmt::format("Illegal column {} of argument of function {}", column->getName(), getName()),
+                ErrorCodes::ILLEGAL_COLUMN);
+    }
+};
+
+class FunctionJsonKeys : public IFunction
+{
+public:
+    static constexpr auto name = "json_keys";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionJsonKeys>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool useDefaultImplementationForNulls() const override { return true; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if unlikely (!arguments[0]->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        return makeNullable(std::make_shared<DataTypeString>());
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const ColumnPtr & column = block.getByPosition(arguments[0]).column;
+        size_t rows = block.rows();
+        if (const auto * col_from = checkAndGetColumn<ColumnString>(column.get()))
+        {
+            const auto & data_from = col_from->getChars();
+            const auto & offsets_from = col_from->getOffsets();
+
+            auto col_to = ColumnString::create();
+            auto & data_to = col_to->getChars();
+            auto & offsets_to = col_to->getOffsets();
+            offsets_to.resize(rows);
+            ColumnUInt8::MutablePtr col_null_map = ColumnUInt8::create(rows, 0);
+            ColumnUInt8::Container & vec_null_map = col_null_map->getData();
+
+            {
+                JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, data_from.size());
```

Comment:
I'm a little worried that it will reserve too many memories for the result.

Reply:
Maybe we can use two loops, first is to extract the keys from orignal json column, then use another loop to do the copy? The intermediate result should not be too large if we keep the key_binaries as StringRef refering the orignal data in `data_from`?

Reply:
This could be inconvenient, especially when dealing with nullmap. 
Typically, keys are very small, how about automatically resize instead of reserve?
https://github.com/pingcap/tiflash/pull/8491/files#diff-52627c7f14d286918db163406a85acac41a1ad79c22184768379ebe163c1131aR1664-R1665